crates/task-impls/src/quorum_proposal_recv/handlers.rs

```diff
@@ -104,6 +106,35 @@ async fn validate_proposal_liveness<TYPES: NodeType, I: NodeImplementation<TYPES
     Ok(())
 }
 
+/// Spawn a task which will fire a request to get a proposal, and store it.
+#[allow(clippy::too_many_arguments)]
+fn spawn_fetch_proposal<TYPES: NodeType, V: Versions>(
+    view: TYPES::View,
+    event_sender: Sender<Arc<HotShotEvent<TYPES>>>,
+    event_receiver: Receiver<Arc<HotShotEvent<TYPES>>>,
+    membership: Arc<TYPES::Membership>,
+    consensus: OuterConsensus<TYPES>,
+    sender_public_key: TYPES::SignatureKey,
+    sender_private_key: <TYPES::SignatureKey as SignatureKey>::PrivateKey,
+    upgrade_lock: UpgradeLock<TYPES, V>,
+) {
+    async_spawn(async move {
+        let lock = upgrade_lock;
+
+        let _ = fetch_proposal(
+            view,
+            event_sender,
+            event_receiver,
+            membership,
+            consensus,
+            sender_public_key,
+            sender_private_key,
+            &lock,
+        )
+        .await;
+    });
```

Comment:
Does this get included with the cancellation logic somehow? Do we want it to be?

Reply:
Sure

Reply:
yeah bounded at 500ms, and doesn't do anything heavy after sending the request

Reply:
would it be bad to add a `async_timeout()` on the entire `fetch_proposal` call? I guess the question is whether it's okay for `fetch_proposal` to abort partway through execution

I agree that it's unlikely `fetch_proposal` will ever get stuck outside the part where we already have an `async_timeout`, but it makes me a bit uncomfortable to drop the handle of tasks that need both a read lock and a write lock (at the very least, I think this might obscure a deadlock if we run into one)